test/extractor/ValueSetFilterComponentRuleExtractor.test.ts

```diff
@@ -0,0 +1,200 @@
+    input.filter = [
+      { property: 'a', op: 'is-a', value: 'b' },
+      { property: 'c', op: 'descendent-of', value: 'd' },
+      { property: 'e', op: 'is-not-a', value: 'f' },
+      { property: 'g', op: 'generalizes', value: 'h' }
+    ];
+    const filter = ValueSetFilterComponentRuleExtractor.process(input, VALUESET, true);
+    const expectedFilter = new ExportableValueSetFilterComponentRule(true);
+    expectedFilter.from = { system: 'http://mycodesystem.org' };
+    expectedFilter.filters = [
+      { property: 'a', operator: fshtypes.VsOperator.IS_A, value: new fshtypes.FshCode('b') },
+      {
+        property: 'c',
+        operator: fshtypes.VsOperator.DESCENDENT_OF,
+        value: new fshtypes.FshCode('d')
+      },
+      { property: 'e', operator: fshtypes.VsOperator.IS_NOT_A, value: new fshtypes.FshCode('f') },
+      { property: 'g', operator: fshtypes.VsOperator.GENERALIZES, value: new fshtypes.FshCode('h') }
+    ];
+    expect(filter).toEqual(expectedFilter);
+  });
+
+  it('should extract regex-based filter rules', () => {
+    const input: fhirtypes.ValueSetComposeIncludeOrExclude = { system: 'http://mycodesystem.org' };
+    input.filter = [{ property: 'a', op: 'regex', value: 'b' }];
+    const filter = ValueSetFilterComponentRuleExtractor.process(input, VALUESET, true);
+    const expectedFilter = new ExportableValueSetFilterComponentRule(true);
+    expectedFilter.from = { system: 'http://mycodesystem.org' };
+    expectedFilter.filters = [
+      { property: 'a', operator: fshtypes.VsOperator.REGEX, value: new RegExp('b') }
+    ];
+    expect(filter).toEqual(expectedFilter);
+  });
+
+  it('should extract boolean-based filter rules', () => {
+    const input: fhirtypes.ValueSetComposeIncludeOrExclude = { system: 'http://mycodesystem.org' };
+    input.filter = [
+      { property: 'a', op: 'exists', value: 'true' },
+      { property: 'c', op: 'exists', value: 'false' }
+    ];
+    const filter = ValueSetFilterComponentRuleExtractor.process(input, VALUESET, true);
+    const expectedFilter = new ExportableValueSetFilterComponentRule(true);
+    expectedFilter.from = { system: 'http://mycodesystem.org' };
+    expectedFilter.filters = [
+      { property: 'a', operator: fshtypes.VsOperator.EXISTS, value: true },
+      { property: 'c', operator: fshtypes.VsOperator.EXISTS, value: false }
+    ];
+    expect(filter).toEqual(expectedFilter);
+  });
+
+  it('should log an error for an unknown operator and treat it as a string-based rule', () => {
+    const input: fhirtypes.ValueSetComposeIncludeOrExclude = { system: 'http://mycodesystem.org' };
+    input.filter = [{ property: 'a', op: 'begets', value: 'b' }];
+    loggerSpy.reset();
+    const filter = ValueSetFilterComponentRuleExtractor.process(input, VALUESET, false);
+    expect(loggerSpy.getLastMessage('error')).toBe(
+      'Unsuported filter operator in ValueSet test-vs: begets'
```

Comment:
Equivalent typo here

```suggestion
      'Unsupported filter operator in ValueSet test-vs: begets'
```